src/payload.rs

```diff
@@ -186,11 +196,35 @@ where
         (&mut self.grant[EsbHeader::dma_payload_offset()..]).as_mut_ptr()
     }
 
+    /// Update the pipe field.
+    ///
+    /// Pipe must be between 0 and 7, inclusive.
+    #[inline]
+    pub(crate) fn set_pipe(&mut self, pipe: u8) {
```

Comment:
For simple fields like this (that are just u8s) we might consider just making these fields `pub(crate)`. We should keep the accessor funcs for the `pid_no_ack` fields though.

Reply:
Hmm, this is working directly on the `Payloadx` though, which doesn't really have an `EsbHeader` field.